### ISSUE TITLE
Are new Zotero beta notes and annotations (not saved in the PDF) retrievable in Pyzotero?

### ISSUE THREAD

**Issue by e-alizadeh** [urschrei/pyzotero#148]

**Comment by urschrei** [comment]
This is really a question for the Zotero Web API maintainers: I'm not aware of any Web API methods that allow retrieval of the annotations. The google group is probably the best place to ask: https://groups.google.com/g/zotero-dev

If there are methods available, I'll add them to Pyzotero.

**Comment by e-alizadeh** [comment]
Thanks for your quick reply. 
I further looked at this issue. Zotero **beta** stores annotations and highlights in their database as opposed to saving them in the PDF files itself. These new annotations are available in Zotero's new PDF editor. 

As per following discussion (https://forums.zotero.org/discussion/90753/zotero-beta-integration-with-external-note-taking-tools), web API access to all annotations made in the new PDF editor is available even now. However, annotations aren't yet documented in the web API docs, even though Zotero itself syncs them via the API as a new item type.

Thanks again

**Comment by urschrei** [comment]
Yep, this is the tracking issue: https://github.com/zotero/dataserver/issues/113.

In the mean time, retrieving annotations using `items(itemType='annotation')` for an initial dump (which you could dump to JSON along with the library version), and then `items(itemType='annotation, since=library_version)` to get updates is a bit convoluted, but will work: retrieved annotations have a `parentItem` key in `data` that you can use for matching. You then _also_ have to rebuild the annotation images from the raw data (easy in case of rects, more complicated for freehand shapes)…

**Comment by e-alizadeh** [comment]
How can I retrieve all annotations? Calling `items(itemType="annotation")` returns 100 items only. Even passing `limit` with a higher number than 100 still returns 100 annotations. 
When calling `count_items()`, I can see that the number increases when I'm adding a new annotations, and I can see the annotation when calling items(itemType="annotation"). However, I'm not able to retrieve older annotations to start with!!

**Comment by urschrei** [comment]
Have you tried wrapping the call in Zotero.everything() as per the docs?

https://pyzotero.readthedocs.io/en/latest/#zotero.Zotero.everything

You can also try to build up a list of lists of annotations using one of the generator methods at

https://pyzotero.readthedocs.io/en/latest/#the-follow-and-everything-methods

And then flatten it.

-- 
steph

**Comment by e-alizadeh** [comment]
Thanks a lot.  `all_annotations = zot.everything(zot.items(itemType="annotation"))` seems to return all annotations.
